lib/src/client.dart

```diff
@@ -2561,16 +2561,16 @@ class Client extends MatrixApi {
       if (room != null) {
         final List<BasicEvent> events = [];
         for (final event in _eventsPendingDecryption) {
-          if (event.event.roomID != roomId) continue;
+          if (event.event.room.id != roomId) continue;
           if (!sessionIds.contains(
-            event.event.content['content']?['session_id'],
+            event.event.content.tryGet<String>('session_id'),
           )) {
             continue;
           }
 
-          final decryptedEvent = await event.event.decrypt(room);
-          if (decryptedEvent.content.tryGet<String>('type') !=
-              EventTypes.Encrypted) {
+          final decryptedEvent =
+              await encryption?.decryptRoomEvent(event.event);
```

Comment:
should we wrap this into a try-catch block so it doesn't affect the for loop in case an exception occurs?

Reply:
decryptRoomEvent() itself is now inside a big try-catch block :)